src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+    private static final String SPECIFIED_CHARACTER = "w";
+    private static final String SEPARATOR = "\\W+";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            for (String line = bufferedReader.readLine();
+                    line != null; line = bufferedReader.readLine()) {
+                for (String word : line.toLowerCase().split(SEPARATOR)) {
+                    if (word.startsWith(SPECIFIED_CHARACTER)) {
+                        stringBuilder.append(word.replace(SEPARATOR, ""))
```

Comment:
```suggestion
                        stringBuilder.append(word))
```
that should work without replace if I'm not mistaken